src/tests/test_cfn.py

```diff
@@ -38,6 +39,47 @@ def test_generate_stack_from_input(self):
         end2end = base.generate_environment_name()
         self.assertEqual(generate_stack_from_input('dummy2', end2end, alt_config='alt-config1'), 'dummy2--%s' % end2end)
 
+    @patch('cfn.local')
+    @patch('buildercore.core.active_stack_names')
+    @patch('buildercore.core.find_ec2_instances')
+    def test_altconfig_name_preserved(self, *args):
+        #import pprint
+
+        # create a random instance id for the 'dummy2' project and use the 'alt-config1' alt-config
+        # see: fixtures/dummy2-project.json
+        instance_id = base.generate_environment_name() # "luke-20191001045227-270172"
+        stackname = generate_stack_from_input('dummy2', instance_id, alt_config='alt-config1') # "dummy2--luke-20191001045222-883274"
+
+        # ensure alt-config is in there and correct
+        current_context = context_handler.load_context(stackname)
+
+        #print('current context')
+        # pprint.pprint(current_context)
+
+        # skip calling update_infrastructure, we just want to test the diff with any changes
+        more_context = {
+            # todo: not working
+            #'rds': {
+            #    'allow-major-version-upgrade': True
```

Comment:
todo: this should be possible but isn't. I thought it was because this 'extra' context is simply being `.update`d over the regular context, overwriting any struct at `rds` already, but it's not happening *at all*.